### DISCUSSION TITLE
Using segmentation masks to track shapes

### DISCUSSION THREAD

**Discussion by enates499** [DeepLabCut/napari-deeplabcut#102] [General]
Is it possible to track shapes rather than points using DLC? I see there is a segmentation mask feature, but is that only for annotations? Even if there was a feature that let you create a segmentation mask, break that up into points around the edges, track those points, and then theoretically you could recreate the shape based off the tracked points at any frame. I am hoping to track a circular shape (or oblong depending on the angle) and would rather not have to place ~16 points around the circle(s) on each frame. Thanks!

**Comment by MMathisLab** [comment]
Its not integrated yet to train a segmentation head within DLC, but we could make a helper function to convert edges to points in napari perhaps! We will have to ask @jeylau nicely :D

**Reply by enates499** [reply]
Thank you! @jeylau if you are able to make this helper function that'd be awesome, or if you have suggestions I can try to implement it on my end. Also curious if either of you know of any segment tracking algorithms - thanks!
